src/odemis/gui/comp/viewport.py

```diff
@@ -1852,6 +1917,62 @@ def _on_new_data(self, data):
         self.Refresh()
 
 
+class AngularSpectrumViewport(TwoDViewPort):
+    """
+    Shows an angular spectrum image from a 2D cdd camera with angle in the vertical axis and
+    wavelength in the horizontal axis.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(AngularSpectrumViewport, self).__init__(*args, **kwargs)
+        self.canvas.markline_overlay.val.subscribe(self._on_overlay_selection)
+
+    def _on_overlay_selection(self, pos):
+        """
+        Called whenever the markline position of the overlay changes, to set
+        the selected wavelength & angle in the stream (so that the SinglePoint
+        projections are updated).
+        """
+        if self._stream:
+            if hasattr(self._stream, "selected_wavelength"):
+                self._stream.selected_wavelength.value = pos[0]
+            if hasattr(self._stream, "selected_angle"):
+                self._stream.selected_angle.value = pos[1]
```

Comment:
If the data is now (still) in rad, then you'd need to convert from deg to radians. ie: `math.radians(pos[1])`. Possibly, you'd need to check that selected_angle.unit is "rad" (because if it's "px" it should not be touched).